### PR TITLE
revised coverity fixes

### PR DESCRIPTION
**Pull request by dariaphoebe**
rebased to current master and without the errant hex_decode fix for kadmin/load

### CODE REVIEW

File: kdc/kx509.c [hunk 1]
```diff
@@ -894,10 +892,12 @@ _kdc_do_kx509(kx509_req_context r)
     /* Authenticate the request (consume the AP-REQ) */
     ret = krb5_kt_resolve(r->context, "HDBGET:", &id);
     if (ret) {
+	char *msg = krb5_get_error_message(r->context, ret);
```
Comment (nicowilliams) [inline kdc/kx509.c]: `krb5_get_error_message()` returns a `const char *`.  It does need freeing, yes, but with `krb5_free_error_message()`, not `free()`.

File: kdc/kx509.c [hunk 2]
```diff
@@ -894,10 +892,12 @@ _kdc_do_kx509(kx509_req_context r)
...
         ret = mk_error_response(r->context, r, 1,
                                 KRB5KDC_ERR_S_PRINCIPAL_UNKNOWN,
                                 "Can't open HDB/keytab for kx509: %s",
-                                krb5_get_error_message(r->context, ret));
+				msg);
+	free(msg);
```
Comment (nicowilliams) [inline kdc/kx509.c]: See above.

File: kdc/kx509.c [hunk 3]
```diff
@@ -974,18 +974,20 @@ _kdc_do_kx509(kx509_req_context r)
     /* Extract and parse CSR or a DER-encoded RSA public key */
     ret = get_csr(r->context, r);
     if (ret) {
+	char *msg = krb5_get_error_message(r->context, ret);
```
Comment (nicowilliams) [inline kdc/kx509.c]: Ditto.

**Comment by nicowilliams** [comment]
Thanks!

**Comment by jaltman** [comment]
This series should not have been merged as the Travis builds failed.

File: kdc/kx509.c [hunk 4]
```diff
@@ -892,10 +892,12 @@ _kdc_do_kx509(kx509_req_context r)
     /* Authenticate the request (consume the AP-REQ) */
     ret = krb5_kt_resolve(r->context, "HDBGET:", &id);
     if (ret) {
+	const char *msg = krb5_get_error_message(r->context, ret);
         ret = mk_error_response(r->context, r, 1,
                                 KRB5KDC_ERR_S_PRINCIPAL_UNKNOWN,
                                 "Can't open HDB/keytab for kx509: %s",
-                                krb5_get_error_message(r->context, ret));
+				msg);
+	krb5_free_error_message(context, msg);
```
Comment (jaltman) [inline kdc/kx509.c]: The first parameter needs to be r->context.

File: kdc/kx509.c [hunk 5]
```diff
@@ -972,18 +974,20 @@ _kdc_do_kx509(kx509_req_context r)
     /* Extract and parse CSR or a DER-encoded RSA public key */
     ret = get_csr(r->context, r);
     if (ret) {
+	const char *msg = krb5_get_error_message(r->context, ret);
         ret = mk_error_response(r->context, r, 3, ret,
-                                "Failed to parse CSR: %s",
-                                krb5_get_error_message(r->context, ret));
+                                "Failed to parse CSR: %s", msg);
+	krb5_free_error_message(context, msg);
```
Comment (jaltman) [inline kdc/kx509.c]: Ditto

File: kdc/kx509.c [hunk 6]
```diff
@@ -972,18 +974,20 @@ _kdc_do_kx509(kx509_req_context r)
...
 
     /* Authorize the request */
     ret = check_authz(r->context, r, cprincipal);
     if (ret) {
+	const char *msg = krb5_get_error_message(r->context, ret);
         ret = mk_error_response(r->context, r, 3, ret,
-                                "Rejected by policy: %s",
-                                krb5_get_error_message(r->context, ret));
+                                "Rejected by policy: %s", msg);
+	krb5_free_error_message(context, msg);
```
Comment (jaltman) [inline kdc/kx509.c]: Ditto

File: kdc/kx509.c [hunk 7]
```diff
@@ -1003,20 +1007,23 @@ _kdc_do_kx509(kx509_req_context r)
...
         int level = 1;
+	const char *msg = krb5_get_error_message(r->context, ret);
 
         if (ret == KRB5KDC_ERR_POLICY)
             level = 4; /* _kdc_audit_trail() logs at level 3 */
         ret = mk_error_response(r->context, r, level, ret,
-                                "Certificate isuance failed: %s",
-                                krb5_get_error_message(r->context, ret));
+                                "Certificate isuance failed: %s", msg);
+	krb5_free_error_message(context, msg);
```
Comment (jaltman) [inline kdc/kx509.c]: Ditto

File: kdc/kx509.c [hunk 8]
```diff
@@ -1003,20 +1007,23 @@ _kdc_do_kx509(kx509_req_context r)
...
     }
 
     ret = encode_cert_and_chain(r->context->hx509ctx, certs, rep.certificate);
     if (ret) {
+	const char *msg = krb5_get_error_message(r->context, ret);
         ret = mk_error_response(r->context, r, 1, ret,
                                 "Could not encode certificate and chain: %s",
-                                krb5_get_error_message(r->context, ret));
+				msg);
+	krb5_free_error_message(context, msg);
```
Comment (jaltman) [inline kdc/kx509.c]: Ditto

**Pull request merged**